### PR TITLE
Update variable names for cost curve visualization

### PR DESCRIPTION
**Pull request by lanesmith**
### Purpose
This change updates the cost curve visualization variable names to be more descriptive.

### What the code is doing
No functionality is changed. The refactored code still offers the ability to visualize and explore generator supply curves.

### Testing
I updated the tests in `powresimdata/design/generation/tests/test_cost_curves.py` and they pass. I also double checked the outputs with an old demo notebook and the outputs remained the same.

### Where to look
The refactor is contained within `powersimdata/design/generation/cost_curves.py` and `powersimdata/design/generation/tests/test_cost_curves.py`. I also updated `README.md` to reflect the changes.

### Time estimate
This should be pretty quick.

### CODE REVIEW

File: powersimdata/design/generation/cost_curves.py [hunk 1]
```diff
@@ -211,60 +218,62 @@ def build_supply_curve(grid, num_segments, area, gen_type, area_type=None, plot=
...
-    P = []  # noqa: N806
-    F = []  # noqa: N806
-    p_diff_sum = 0
-    for i in df.index:
-        P.append(p_diff_sum)
-        F.append(df["slope"][i])
-        P.append(df["p_diff"][i] + p_diff_sum)
-        F.append(df["slope"][i])
-        p_diff_sum += df["p_diff"][i]
+    capacity_data = []  # noqa: N806
```
Comment (jon-hagg) [inline powersimdata/design/generation/cost_curves.py:262]: I think we can remove all the `noqa` comments on the lines you changed, now that the naming follows pep8.

Comment (lanesmith) [reply powersimdata/design/generation/cost_curves.py:262]: Okay I can remove those. I wasn't sure what the `noqa` comments meant.

Comment (lanesmith) [reply powersimdata/design/generation/cost_curves.py]: Done.

Comment (jon-hagg) [reply powersimdata/design/generation/cost_curves.py]: Thanks!

**Review by jon-hagg — APPROVED** [review]

**Pull request merged**